crates/rattler/src/package_cache.rs

```diff
@@ -62,13 +65,49 @@ struct Package {
 }
 
 /// Required information about a package we want to retrieve or store in the cache.
-#[derive(Debug)]
+#[derive(Clone, Debug, Hash, PartialEq, Eq)]
 pub struct PackageInfo {
     pub name: String,
     pub version: String,
     pub build_string: String,
 }
 
+impl Display for PackageInfo {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}-{}-{}", &self.name, &self.version, &self.build_string)
+    }
+}
+
+impl PackageInfo {
+    /// Try to convert the specified filename into a [`PackageInfo`].
+    pub fn try_from_filename(filename: &str) -> Option<PackageInfo> {
+        // Filenames in the form of: <name>-<version>-<build>(.ext)
+        let (build_string, version, name) = filename.rsplitn(3, '-').next_tuple()?;
+
+        // Remove the file extension from the build string
+        let build_string = build_string
```

Comment:
packages can have `.` in names. I think we shoudl always match `.conda` or `.tar.bz2` explicitly.

Reply:
But can build strings contain strings? At this point in the code the only the build string part is split.